olive/data/config.py

```diff
@@ -77,86 +58,74 @@ def __init__(self, **kwargs):
-                self.components[k] = deepcopy(v)
-            else:
-                # both are strings, so we don't need to deepcopy
-                self.components[k].type = self.components[k].type or v.type
+            if components[k]:
+                # type is string, so we don't need to deepcopy
+                components[k].type = components[k].type or v.type
                 # v.params is a dict, so we deepcopy it
-                self.components[k].params = self.components[k].params or deepcopy(v.params)
+                components[k].params = components[k].params or deepcopy(v.params)
+            else:
+                # v is a DataComponentConfig object, so we deepcopy it
+                components[k] = deepcopy(v)
+
+        self.load_dataset_config = components[DataComponentType.LOAD_DATASET.value]
+        self.pre_process_data_config = components[DataComponentType.PRE_PROCESS_DATA.value]
+        self.post_process_data_config = components[DataComponentType.POST_PROCESS_DATA.value]
+        self.dataloader_config = components[DataComponentType.DATALOADER.value]
 
-    def _update_default_component_type(self):
+    def _get_default_components(self):
         """Resolve the default component type."""
+        # 1. get default_components_type from DataContainer or DefaultDataComponentCombos
         dc_cls = Registry.get_container(self.type)
         # deepcopy dc_cls.default_components_type since we don't want to update dc_cls.default_components_type
-        self.default_components_type = deepcopy(dc_cls.default_components_type) or {}
-        # 1. update default_components_type with task_type for huggingface case
-        self._update_default_component_type_with_task_type(dc_cls)
-        # 2. update default_components_type with DefaultDataComponentCombos
+        default_components_type = deepcopy(dc_cls.default_components_type) or {}
+        # update default_components_type with task_type for huggingface case
+        self._update_default_component_type_with_task_type(dc_cls, default_components_type)
+        # update default_components_type with DefaultDataComponentCombos
         # for those components not defined in the container config
         for k, v in DefaultDataComponentCombos.items():
-            if k not in self.default_components_type:
-                self.default_components_type[k] = v
+            if k not in default_components_type:
+                default_components_type[k] = v
 
-    def _update_default_component(self):
-        """Resolve the default component type."""
-        for k, v in self.default_components_type.items():
-            self.default_components[k] = DataComponentConfig(type=v, params={})
+        # 2. get default_components from default_components_type
+        return {k: DataComponentConfig(type=v, params={}) for k, v in default_components_type.items()}
 
-    def fill_in_params(self):
+    def _fill_in_params(self):
```

Comment:
I don't think this is needed anymore? If the function as default values, then we don't need to add it to `params` which is passed as `kwargs` anyways.

Reply:
That's change is going to require a full suite of tests again. Can I remove the call in a follow up PR?

Reply:
sure! Doesn't affect the behavior so it's not high priority. just think it's redundant.

Reply:
Done.

Reply:
Apparently, this is still required. The call appends keys to the `params` that are initialized as `None` and will potentially be filled in later. Few tests failed if I removed this call -

https://aiinfra.visualstudio.com/PublicPackages/_build/results?buildId=480801&view=logs&j=dbbdd4d8-ffd3-515a-31fd-659375dbdff8&t=0a2248d7-6ef4-5599-1bc5-d55a003ee0aa

Change reverted.